### PR TITLE
[ETCM-123] Add metrics related to network, rpc and mempool

### PR DESCRIPTION
**Pull request by mmrozek**
# Description

Added metrics:
- network (number of handshaked/pending/discovered/blaclisted peers, number of sent/received msgs)
- mempool (number of pending txs)
- rpc (number of method finished sucessfully/failed, times of handling request)

### CODE REVIEW

**Review by pslaski — APPROVED** [review]

File: src/main/scala/io/iohk/ethereum/network/PeerManagerActor.scala [hunk 1]
```diff
@@ -92,17 +93,21 @@ class PeerManagerActor(
...
 
       val nodesToConnect = nodesInfo
-        .filterNot{ discoveryNodeInfo =>
+        .filterNot { discoveryNodeInfo =>
           val socketAddress = discoveryNodeInfo.node.tcpSocketAddress
           peerAddresses.contains(socketAddress) || isBlacklisted(PeerAddress(socketAddress.getHostString))
         } // not already connected to or blacklisted
         .take(peerConfiguration.maxOutgoingPeers - peerAddresses.size)
 
+      NetworkMetrics.DiscoveredPeersSize.set(nodesInfo.size)
```
Comment (ntallar) [inline src/main/scala/io/iohk/ethereum/network/PeerManagerActor.scala:102]: Aren't `nodes` the discovered ones? They have been some of them were removed when getting `nodesInfo`

File: src/main/scala/io/iohk/ethereum/network/PeerManagerActor.scala [hunk 2]
```diff
@@ -92,17 +93,21 @@ class PeerManagerActor(
...
+      NetworkMetrics.BlacklistedPeersSize.set(blacklistedPeers.size)
+      NetworkMetrics.PendingPeersSize.set(nodesToConnect.size)
```
Comment (ntallar) [inline src/main/scala/io/iohk/ethereum/network/PeerManagerActor.scala:104]: Why using nodesToConnect here and not `pendingPeers`?

**Review by ntallar — CHANGES_REQUESTED** [review]

File: src/main/scala/io/iohk/ethereum/network/PeerManagerActor.scala (hunk 2 above)
Comment (mmrozek) [reply src/main/scala/io/iohk/ethereum/network/PeerManagerActor.scala:104]: You're right. I will fix that

File: src/main/scala/io/iohk/ethereum/network/PeerManagerActor.scala (hunk 1 above)
Comment (ntallar) [reply src/main/scala/io/iohk/ethereum/network/PeerManagerActor.scala:102]: Oh you're right, I misread

**Review by ntallar — APPROVED** [review]
LGTM!
```
 .\\            //.
. \ \          / /.
.\  ,\     /` /,.-
 -.   \  /'/ /  .
 ` -   `-'  \  -
   '.       /.\`
      -    .-
      :`//.'
      .`.'
      .' BP 
```

**Pull request merged**